library-model/library-model-generator/src/main/java/com/uber/nullaway/libmodel/MethodAnnotationsRecord.java

```diff
@@ -0,0 +1,20 @@
+package com.uber.nullaway.libmodel;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+/** A record describing the annotations associated with a java method and its arguments. */
+@AutoValue
```

Comment:
Are we otherwise using `@AutoValue` for something or what's the advantage of it here?

Reply:
Sorry this is my fault.  I thought this was new code and thought that `@AutoValue` made it a bit cleaner.  I didn't realize it had been moved from somewhere else.  @lazaroclapp would you like us to undo the `@AutoValue`-ization of this file?

Reply:
Never mind, seems we have been using `@AutoValue` in at least some parts of the codebase, since the [beginning](https://github.com/uber/NullAway/commit/88bdd4ae241440d9cc385bffd283d6883e539e0b#diff-80e3e738d2c42a3ffa43726169601225c70fa9ab18766db23c64a9dee80efd19R216), so I am fine having it here too, at least until we can do JDK 17 records.